tests/test_bot.py

```
@@ -0,0 +1,931 @@
+            mock_event.sender = "@user:matrix.org"  # Different from bot
+            mock_event.server_timestamp = 2000  # After start time
+            mock_event.body = "John 3:16"
+            mock_event.event_id = "$event:matrix.org"
+
+            # Mock the scripture handling
+            with patch.object(bot_instance, "handle_scripture_command") as mock_handle:
+                await bot_instance.on_room_message(mock_room, mock_event)
+
+                mock_handle.assert_called_once()
+                call_args = mock_handle.call_args[0]
+                assert call_args[0] == "!room1:matrix.org"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)